### PR TITLE
Create AppStorePrice component

### PR DESCRIPTION
**Pull request by pedromtec**
#### What does this PR do? \*
Create first version of `AppStorePrice` component
#### How to test it? \*
[workspace](https://pricenew--extensions.myvtex.com/google-shopping/p)
You can check others apps too. 

#### Related to / Depends on \*
[app-store-api](https://github.com/vtex/app-store-api/pull/7)

### CODE REVIEW

File: react/AppStorePrice.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import { useFormattedPrice } from 'vtex.formatted-price'
+import { AppPrice, Metric } from 'vtex.app-store-api'
+import useProduct from 'vtex.product-context/useProduct'
+import { Spinner } from 'vtex.styleguide'
+import { useIntl, defineMessages } from 'react-intl'
+import { useCssHandles } from 'vtex.css-handles'
+
+import APP_PRICE from './graphql/queries/AppPrice.graphql'
+
+const CSS_HANDLES = ['container', 'listItem', 'itemListFirst', 'itemListSecond']
```
Comment (arturpimentel) [inline react/AppStorePrice.tsx:12]:
```suggestion
const CSS_HANDLES = ['container', 'listItem', 'itemListFirst', 'itemListSecond'] as const
```

File: react/AppStorePrice.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+const AppStorePrice: FunctionComponent = () => {
+  const {
+    product: { productId },
+  } = useProduct()
+
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const { data, loading } = useQuery<AppPriceQuery>(APP_PRICE, {
```
Comment (arturpimentel) [inline react/AppStorePrice.tsx:144]: You may want to use the types generated by `vtex setup` to variables as well
```suggestion
  const { data, loading } = useQuery<AppPriceQuery, AppPriceQueryVariables>(APP_PRICE, {
```

File: react/AppStorePrice.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    <Fragment>
+      <span className={`${handles.itemListFirst} mid-gray w-60`}>
+        {description}
+      </span>
+      <span className={`${handles.itemListSecond} mid-gray w-40 f5`}>
+        {formattedValue}
+      </span>
+    </Fragment>
+  )
+}
```
Comment (arturpimentel) [inline react/AppStorePrice.tsx:60]: These components could use better names, such as:
```suggestion
const ListItemContainer: FunctionComponent<{ paddingTop?: boolean }> = ({
  children,
  paddingTop,
}) => {
  const handles = useCssHandles(CSS_HANDLES)
  return (
    <div
      className={`${handles.listItem} flex bb b--muted-4 pb6 ${
        paddingTop ? 'pt6' : ''
      }`}
    >
      {children}
    </div>
  )
}

const ListItemContent: FunctionComponent<{
  description: string
  formattedValue: string
}> = ({ description, formattedValue }) => {
  const handles = useCssHandles(CSS_HANDLES)
  return (
    <Fragment>
      <span className={`${handles.itemListFirst} mid-gray w-60`}>
        {description}
      </span>
      <span className={`${handles.itemListSecond} mid-gray w-40 f5`}>
        {formattedValue}
      </span>
    </Fragment>
  )
}
```

File: react/package.json [hunk 4]
```diff
@@ -21,6 +21,11 @@
     "apollo-cache-inmemory": "^1.6.5",
     "graphql": "^14.6.0",
     "prop-types": "^15.7.2",
-    "typescript": "3.8.3"
+    "typescript": "3.8.3",
+    "vtex.app-store-api": "https://pricenew--extensions.myvtex.com/_v/private/typings/linked/v1/vtex.app-store-api@0.2.1+build1594660786/public/@types/vtex.app-store-api",
```
[the remark this answers is not in the document]
Comment (pedromtec) [reply react/package.json:25]: Nope

**Pull request merged**